includes/formatting.php

```diff
@@ -45,7 +49,70 @@ function give_get_price_thousand_separator() {
  * @return mixed
  */
 function give_get_price_decimal_separator() {
-	return give_get_option( 'decimal_separator', '.' );
+	$default_decimal_separators = array(
+		'.' => ',',
+		',' => '.',
+	);
+
+	$thousand_separator = give_get_price_thousand_separator();
+	$default_decimal_separator = in_array( $thousand_separator, $default_decimal_separators ) ?
+		$default_decimal_separators[$thousand_separator] :
+		'.';
+
+	$decimal_separator = give_get_option( 'decimal_separator', $default_decimal_separator );
+
+	return $decimal_separator;
+}
+
+
+/**
+ * Sanitize Amount before saving to database
+ *
+ * @since      1.8.12
+ *
+ * @param  int|float|string $number Expects either a float or a string with a decimal separator only (no thousands)
+ *
+ * @return string $amount Newly sanitized amount
+ */
+function give_sanitize_amount_for_db( $number ) {
+	return give_sanitize_amount( $number, 6 );
+}
+
+/**
+ * Sanitize Amount before saving to database
+ *
+ * @since      1.8.12
+ *
+ * @param  int|float|string $number     Expects either a float or a string with a decimal separator only (no thousands)
+ * @param  int|bool         $dp         Number of decimals
+ * @param  bool             $trim_zeros From end of string
+ *
+ * @return string $amount Newly sanitized amount
+ */
+function give_maybe_sanitize_amount( $number, $dp = false, $trim_zeros = false ) {
```

Comment:
@DevinWalker Helper function which used inside the core. It will decide whom to sanitize.